### PR TITLE
Stop block transpilation after encountering a return statement

### PR DESCRIPTION
**Pull request by Perryvw**
Addresses #153

### CODE REVIEW

File: src/Transpiler.ts [hunk 1]
```diff
@@ -224,7 +224,17 @@ export abstract class LuaTranspiler {
...
         this.exportStack.push([]);
-        let result = block.statements.map(statement => this.transpileNode(statement)).join("");
+
+        let result = "";
+        for (const statement of block.statements) {
+            result += this.transpileNode(statement);
+
+            // Don't transpile any dead code after a return
+            if (ts.isReturnStatement(statement)) {
+                break;
```
Comment (lolleko) [inline src/Transpiler.ts]: Please no breaks.

File: test/unit/functions.spec.ts [hunk 2]
```diff
@@ -204,4 +204,20 @@ export class FunctionTests {
...
+        Expect(result).toBe(3);
+    }
+
+    @Test("Method dead code after return")
+    public methodDeadCodeAfterReturn(): void {
+        const result = util.transpileAndExecute(
+            `class def { public static abc() { return 3; const a = 5; } } return def.abc();`);
+
+        Expect(result).toBe(3);
+    }
```
Comment (lolleko) [inline test/unit/functions.spec.ts]: namespaces also use transpire block a test would be nice.

File: test/unit/conditionals.spec.ts [hunk 3]
```diff
@@ -251,4 +251,12 @@ export class LuaConditionalsTests {
...
         Expect(result).toBe(expected);
     }
+
+    @Test("If dead code after return")
+    public ifDeadCodeAfterReturn(): void {
+        const result = util.transpileAndExecute(
+            `if (true) { return 3; const b = 8; }`);
+
+        Expect(result).toBe(3);
+    }
```
Comment (lolleko) [inline test/unit/conditionals.spec.ts]: switch also use transpileblock irrc, but I think this is less important than the namespace test and probably not needed.

File: test/unit/functions.spec.ts (hunk 2 above)
Comment (Perryvw) [reply test/unit/functions.spec.ts]: You can not return from a namespace since it is not a function body.

**Pull request merged**